OpenRA.Mods.Common/Scripting/Global/ReinforcementsGlobal.cs

```diff
@@ -146,14 +148,32 @@ public Actor[] Reinforce(Player owner, string[] actorTypes, CPos[] entryPath, in
 				var aircraftInfo = transport.TraitOrDefault<Aircraft>();
 				if (aircraftInfo != null)
 				{
+					var destination = entryPath.Last();
+
+					// Try to find an alternative landing spot if we can't land at the current destination
+					if (!aircraftInfo.CanLand(destination) && dropRange > 0)
+					{
+						foreach (var c in transport.World.Map.FindTilesInCircle(destination, dropRange))
+						{
+							if (!aircraftInfo.CanLand(c))
```

Comment:
Please also do a `DomainIndex` check to make sure that this isn't on the wrong side of a river/cliff/other obstacle.